src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,147 @@
+package duke.parser;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import duke.DukeException;
+import duke.InvalidArgumentException;
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Responsible for parsing user input and identifying correctly formatted commands.
+ */
+public class Parser {
+
+    /**
+     * Parses and identifies the command provided by the user.
+     *
+     * @param strCommand The user's command in String format.
+     * @return The Command object corresponding to the user's input.
+     * @throws DukeException If the user provides an invalid or incorrectly formatted command.
+     */
+    public static Command parse(String strCommand) throws DukeException {
+        String[] splitCommand = strCommand.split(" ", 2);
+        String keyword = splitCommand[0];
+        Command command;
+
+        switch (keyword) {
+        case "bye":
+            command = new ExitCommand();
```

Comment:
Good job configuring the `switch` statement's indentation to fit the coding standard!